ls1intum/Artemis
# `Exam mode`: Fix missing repository lock notifications

### Checklist
#### General
#### Server
#### Client

### Motivation and Context
The "The student repositories for this programming exercise were locked successfully." notification is missing for exam exercises, but it is an important piece of information for instructors to verify if the operation before and after the exam went well.

### Description
The issue was that these type of notifications were "exercise updates" before. In exam exercises, these notifications get filtered out of from the notification bar (since the notification bar is not present during an exam and updates directly get highlighted in the exam viewer).

This PR introduces a new notification type for repository permission updates, which therefore won't be filtered out in the notification bar.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Programming Exercise

1. Press the "lock repositories" / "unlock repositories" buttons and check that a notification is present
2. Wait until after the due date and check that an automatic notification is present.

#### Exam Mode Testing
Prerequisites:
- 1 Instructor
- 1 Exam with a Programming Exercise

1. Press the "lock repositories" / "unlock repositories" buttons and check that a notification is present
2. Wait until after the exam and check that an automatic notification is present.

### Review Progress

#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage

### Screenshots
